### ISSUE TITLE
Issue with AttributeError: 'Config' object has no attribute 'jax_experimental_name_stack'

### ISSUE THREAD

**Issue by catslikefelixs** [zephyris/discoba_alphafold#2]
Hi there,

Apologies in advance if this issue hasn't enough detail.

I'm trying to run the example query sequence (A4IBK2) and few hypothetical sequences and I seem to be running up against a consistent Attribute Error in the 8th cell - "Gather input features, predict structure":

"running model_1

---------------------------------------------------------------------------

UnfilteredStackTrace                      Traceback (most recent call last)

[<ipython-input-23-4882f4ef7f6e>](https://localhost:8080/#) in <module>
     60                          model_params=model_params, use_model=use_model,
---> 61                          do_relax=use_amber)

28 frames

UnfilteredStackTrace: AttributeError: 'Config' object has no attribute 'jax_experimental_name_stack'

The stack trace below excludes JAX-internal frames.
The preceding is the original exception that occurred, unmodified.

--------------------

The above exception was the direct cause of the following exception:

AttributeError                            Traceback (most recent call last)

[/usr/local/lib/python3.7/dist-packages/haiku/_src/module.py](https://localhost:8080/#) in wrapped(self, *args, **kwargs)
    419       f = functools.partial(unbound_method, self)
    420       f = functools.partial(run_interceptors, f, method_name, self)
--> 421       if jax.config.jax_experimental_name_stack and module_name:
    422         local_module_name = module_name.split("/")[-1]
    423         f = jax.named_call(f, name=local_module_name)

AttributeError: 'Config' object has no attribute 'jax_experimental_name_stack'

From looking at Stack Overflow the issue seems to be with the version of JAX. I have tried the following recommendation but this does not seem to fix the issue:

"jax.tree_multimap was [deprecated in JAX version 0.3.5](https://jax.readthedocs.io/en/latest/changelog.html#jax-0-3-5-april-7-2022), and [removed in JAX version 0.3.16](https://jax.readthedocs.io/en/latest/changelog.html#jax-0-3-16).

You can either change the source to use jax.tree_map as a drop-in replacement for jax.tree_multimap, or install an older version of JAX, e.g.:

!pip install "jax<=0.3.16" "jaxlib<=0.3.16"

And then be sure to restart your runtime to pick up the new version."

I had tried updating the code to install version 0.3.16 of JAX but to no avail. I can't seem to find where in the code to swap the jax.tree_multimap for jax.tree_map, unfortunately. Any help or advice would be greatly appreciated. Thanks in advance, Felix.

**Comment by efrutos** [comment]
Hi there, 
When I try to run any protein query, even the one used as example, I obtain the following error list, and I don´t know how to fix them. Any help? 
Thank you very much

ModuleNotFoundError                       Traceback (most recent call last)
[<ipython-input-22-87210b6490cb>](https://localhost:8080/#) in <module>
     19   from alphafold.data import pipeline
     20   from alphafold.data import templates
---> 21   from alphafold.model import data
     22   from alphafold.model import config
     23   from alphafold.model import model

9 frames
[/usr/local/lib/python3.8/dist-packages/jax/_src/lib/mlir/dialects/__init__.py](https://localhost:8080/#) in <module>
     17 import jaxlib.mlir.dialects.chlo as chlo
     18 import jaxlib.mlir.dialects.mhlo as mhlo
---> 19 import jaxlib.mlir.dialects.std as std

ModuleNotFoundError: No module named 'jaxlib.mlir.dialects.std'


Edit: According to this, I found next error in "Install dependencies" window: 
ERROR: jaxlib-0.1.76+cuda11.cudnn805-cp37-none-manylinux2010_x86_64.whl is not a supported wheel on this platform.

**Comment by zephyris** [comment]
This is a known issue I have been looking into, which is unfortunately non-trivial to solve. There is a complex dependency on jaxlib, tensorflow, cuda and ubuntu versions which is hard to control on Google colab.

As an interim solution, I suggest using the alignment generation notebooks (https://github.com/zephyris/discoba_alphafold/blob/main/DiscobaHMMER.ipynb) to make a a3m and put that into the standard colabfold (https://github.com/sokrypton/Colabfold) as a custom a3m alignment file. 

The plan is to rewrite these colab notebooks completely to use the new colabfold codebase to address these issues.

**Comment by zephyris** [comment]
I've worked up an experimental notebook (https://github.com/zephyris/discoba_alphafold/blob/main/DiscobaAlphaFold2HMMERv2.ipynb) which _should_ work as expected. I also took this chance to incorporate the amber relaxation, PDB-templated, recycles features and more of the latest ColabFold versions. This will be the one I update moving forward.

As a technical warning, the MMSeqs API is currently unavailable. Instead this notebook relies on a hack to get the standard ColabFold MMSeqs alignment via a deliberately broken (0 predictions) ColabFold prediction request. This may well break in the future...
